templates/GenericTemplate.hs

```diff
@@ -139,22 +139,16 @@ happyDoAction i tk st
 #endif /* HAPPY_ARRAY */
 
 #ifdef HAPPY_GHC
-indexShortOffAddr (HappyA# arr) off =
-        Happy_GHC_Exts.narrow16Int# i
-  where
-        i = Happy_GHC_Exts.word2Int# (Happy_GHC_Exts.or# (Happy_GHC_Exts.uncheckedShiftL# high 8#) low)
-        high = Happy_GHC_Exts.int2Word# (Happy_GHC_Exts.ord# (Happy_GHC_Exts.indexCharOffAddr# arr (off' Happy_GHC_Exts.+# 1#)))
-        low  = Happy_GHC_Exts.int2Word# (Happy_GHC_Exts.ord# (Happy_GHC_Exts.indexCharOffAddr# arr off'))
-        off' = off Happy_GHC_Exts.*# 2#
+indexShortOffAddr (HappyA# arr) off = Happy_GHC_Exts.indexInt16OffAddr# arr off
```

Comment:
Why is this changed? I believe the code here is to make the index endian-independent.

Reply:
I've changed this back - your comment is obvious in hindsight.